### PR TITLE
Clang warnings

### PR DESCRIPTION
**Pull request by PaulWagener**
Fixes all Clang warnings

Warnings: https://travis-ci.org/bliksemlabs/rrrr/jobs/51539959
Warnings fixed: https://travis-ci.org/bliksemlabs/rrrr/jobs/51593842

Also removed a bunch of dead code

### CODE REVIEW

**Comment by PaulWagener** [comment]
Thanks for the feedback, these commits should take care of most of the issues.

File: router_request.c [hunk 1]
```diff
@@ -317,15 +317,15 @@ router_request_reverse_all(router_t *router, router_request_t *req, router_reque
     } else {
         best_sp_index = (req->arrive_by ? req->from_stop_point : req->to_stop_point);
         do {
-            best_time = router->states_time[round * router->tdata->n_stop_points + best_sp_index];
+            best_time = router->states_time[(unsigned int) round * router->tdata->n_stop_points + best_sp_index];
```
Comment (skinkie) [inline router_request.c]: we don't use "unsigned int": i suggest,

router->tdata->n_stop_points \* round + best_sp_index

Comment (PaulWagener) [inline router_request.c]: That doesn't solve the warning that `round` (a signed variable) is used in an array index calculation (which must always result in an unsigned value)

Comment (PaulWagener) [inline router_request.c]: My first solution was to change `round` to uint8_t, but that messes up the do-while loop condition at line 316

Comment (skinkie) [inline router_request.c]: For now make it uint8_t

Comment (skinkie) [inline router_request.c]: router->tdata->n_stop_points \* (uint8_t) round + best_sp_index

**Pull request merged**